modules/sku/src/AcquiaCommerce/SKUPluginBase.php

```diff
@@ -318,14 +318,23 @@ public function getProcessedStock(SKU $sku, $reset = FALSE) {
    *
    * @throws \Exception
    */
-  protected function getStock(SKU $sku, $reset = FALSE) {
-    $stock = NULL;
+  protected function getStock($sku, $reset = FALSE) {
     $stock_mode = \Drupal::config('acm_sku.settings')->get('stock_mode');
+    $sku_string = ($sku instanceof SKU) ? $sku->getSku() : $sku;
 
     if (!$reset) {
       // Return from Entity field in push mode.
       if ($stock_mode == 'push') {
-        $stock = $sku->get('stock')->getString();
+        if ($sku instanceof SKU) {
+          $stock = $sku->get('stock')->getString();
+        }
+        else {
+          $stock = \Drupal::database()->select('acm_sku_field_data', 'asfd')
```

Comment:
Please inject \Drupal::database through constructor.

Reply:
done (for another place in same file as well, IMO we should create public tasks to fix such issues in existing code)